internal/core/data/v2/io/event.go

```diff
@@ -8,19 +8,52 @@ package io
 import (
 	"encoding/json"
 	"io"
+	"io/ioutil"
+	"strings"
 
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
 	dto "github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/requests"
+	"github.com/fxamacker/cbor/v2"
 )
 
+// To avoid large data causing unexpected memory exhaustion when decoding CBOR payload, maxEventSize was introduced as
+// a reasonable limit appropriate for handling CBOR payload in edgex-go.  More details could be found at
+// https://github.com/edgexfoundry/edgex-go/issues/2439
+const maxEventSize = int64(25 * 1e6) // 25 MB
```

Comment:
Do we need to make this a service configuration setting? i.e. some systems may need this to be lower while others may handle it being larger?

If so please add a TODO here and an issue to make it a setting.

Reply:
updated https://github.com/edgexfoundry/edgex-go/pull/3202/commits/a97e7d088cfea97e2f7e544ceb56995fb0b64fad